crates/topos-tce-storage/src/fullnode/mod.rs

```diff
@@ -260,14 +287,50 @@ impl ReadStore for FullNodeStore {
 
     fn get_source_stream_certificates_from_position(
         &self,
-        from: SourceStreamPositionKey,
+        from: CertificateSourceStreamPosition,
         limit: usize,
     ) -> Result<Vec<(CertificateDelivered, CertificateSourceStreamPosition)>, StorageError> {
-        let starting_position = from.1;
+        let starting_position = from.position;
         let x: Vec<(CertificateId, CertificateSourceStreamPosition)> = self
             .perpetual_tables
             .streams
-            .prefix_iter(&from.0)?
+            .prefix_iter(&from.subnet_id)?
+            .skip((starting_position.0).try_into().map_err(|_| {
+                StorageError::InternalStorage(InternalStorageError::InvalidQueryArgument(
+                    "Unable to parse Position",
+                ))
+            })?)
+            .take(limit)
+            .map(|(k, v)| (v, k))
+            .collect();
+
+        let certificate_ids: Vec<_> = x.iter().map(|(k, _)| k).cloned().collect();
+
+        let certificates = self
+            .perpetual_tables
+            .certificates
+            .multi_get(&certificate_ids[..])?;
+
+        Ok(x.into_iter()
+            .zip(certificates)
+            .filter_map(|((certificate_id, position), certificate)| {
+                certificate
+                    .filter(|c| c.certificate.id == certificate_id)
+                    .map(|cert| (cert, position))
+            })
+            .collect())
+    }
+
+    fn get_target_stream_certificates_from_position(
+        &self,
+        position: TargetStreamPositionKey,
```

Comment:
Why not using here `CertificateTargetStreamPosition` same as `SourceStreamPositionKey` was removed?